packages/x-date-pickers/src/internals/components/PickersPopper.tsx

```diff
@@ -394,9 +399,12 @@ export function PickersPopper(inProps: PickerPopperProps) {
           disableRestoreFocus
           disableEnforceFocus={role === 'tooltip'}
           isEnabled={() => true}
-          {...componentsProps?.desktopTrapFocus}
+          {...(slotsProps?.desktopTrapFocus ?? componentsProps?.desktopTrapFocus)}
```

Comment:
I do really think that for such use cases the `const slotsProps = innerSlotsProps ?? componentsProps;` pattern would allow for cleaner end result. 🤔